cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -1373,14 +1368,27 @@
 
 ;;;; Helpers for determining the register class of a value type ;;;;;;;;;;;;;;;;
 
+(type RegisterClass
+      (enum
+        (Gpr (single_register bool))
```

Comment:
Can we have SingleGpr and MultiGpr here instead?

Reply:
Without or patterns, that makes the existing behavior of `is_gpr_type` impossible to capture. Is it the case that everywhere it's used now it implies the single register case?

Reply:
Can we use two rules then?

```lisp
(rule (is_gpr_type (RegisterClass.SingleGpr)) $true)
(rule (is_gpr_type (RegisterClass.MultiGpr)) $true)
(rule (is_gpr_type (RegisterClass.Xmm)) $false)
```

In general I'm not a huge fan of isolated bool flags in data structures as the meaning isn't carried with them (one has to look back to the type definition) but if it isn't possible to do here for some reason I'm fine with the original too.

Reply:
Urgh, sorry, just realized (with your pointing out) that this is an extractor definition; in that case the original is fine!